### ISSUE TITLE
Action[SetPoint] failed because [SetPoint] would result in anchor family connection

### ISSUE THREAD

**Issue by Yanngo** [Nevcairiel/Mapster#4]
Addon version: 1.8.4.
Release type: Beta.
As soon as I open the world map, this error appears in BugSack.

> 975x AtlasLoot\Core\WorldMapIntegration.lua:75: Action[SetPoint] failed because[SetPoint would result in anchor family connection]: attempted from: MapsterOptionsButton:SetPoint.
> [C]: in function `SetPoint'
> AtlasLoot\Core\WorldMapIntegration.lua:75: in function <AtlasLoot\Core\WorldMapIntegration.lua:34>
> [C]: in function `Show'
> FrameXML\UIParent.lua:3205: in function <FrameXML\UIParent.lua:3195>
> [C]: in function `ShowUIPanel'
> FrameXML\QuestLogOwnerMixin.lua:103: in function `SetDisplayState'
> FrameXML\QuestLogOwnerMixin.lua:41: in function `HandleUserActionToggleSelf'
> ...rfaceBlizzard_WorldMap\Blizzard_WorldMap.lua:413: in function `ToggleWorldMap'
> [string "TOGGLEWORLDMAP"]:1: in function <[string "TOGGLEWORLDMAP"]:1>
> 
> Locals:
> (*temporary) = MapsterOptionsButton {
>  0 = <userdata>
>  Text = MapsterOptionsButtonText {
>  }
>  Right = <unnamed> {
>  }
>  Middle = <unnamed> {
>  }
>  Left = <unnamed> {
>  }
> }
> (*temporary) = "TOPRIGHT"
> (*temporary) = nil
> (*temporary) = "TOPRIGHT"
> (*temporary) = -20
> (*temporary) = -3

**Comment by Yanngo** [comment]
Oh man. I am dumb. It is AtlasLoot's fault. So sorry.

**Comment by Yanngo** [comment]
On the other hand, could you please look into this? Because it seems the author of AtlasLoot doesn't want to. That issue has been opened since 9 July. Maybe buttons generated by Mapster and AtlasLoot interfere with each other?

**Comment by Nevcairiel** [comment]
I looked at what AtlasLoot is doing, and I can safely say its squarely their problem.

In short, AtlasLoot tries to move a variety of buttons that other addons add to the map, presumably to make room for its own stuff. The problem is that the code to move the Mapster button is extremely out of date and broken (ie. it references WorldMapTitleButton which ceased to exist in WoW 8.0). Additionally, to avoid this particular error, they should always call ClearAllPoints on frames they move.

Anyway, in summary, there is nothing I can do to fix it, they have rather outdated code in there that needs fixing.

As a work-around, the only thing I could possibly do, is adjust it so that when you turn off the button in Mapster it ensures that it doesn't exist at all, so that AtlasLoot can also not mess it up. So when you grab the latest alpha release, and disable the button in the options,  AtlasLoot hopefully won't error on it. Regardless, its their bug to really fix.
